optuna/_deprecated.py

```diff
@@ -1,17 +1,23 @@
 import functools
-import inspect
 import textwrap
 from typing import Any
 from typing import Callable
 from typing import Optional
+from typing import TypeVar
 import warnings
 
 from packaging import version
+from typing_extensions import ParamSpec
```

Comment:
does this break for older versions of typing_extensions, see https://stackoverflow.com/questions/71944041/using-python-code-typed-with-paramspec-on-older-versions-of-python